pyramid/authentication.py

```diff
@@ -1062,6 +1062,16 @@ def __init__(self, prefix='auth.', callback=None, debug=False):
 
     def remember(self, request, userid, **kw):
         """ Store a userid in the session."""
+
+        # To prevent session fixation attacks, we'll copy all of the data from
+        # the session, invalidate the session, and then restore the data. This
+        # should ensure we have a completely new session, ideally with a new
+        # session identifier where applicable, which will protect against a
+        # fixated session crossing authentication boundaries.
+        data = request.session.copy()
```

Comment:
`ISession` extends `IDict` and neither has an explicit requirement that they have a `.copy()` function. It also looks like `IDict` was originally only intended for renderers, looking at the docstring on `IDict.update(d)`.